### ISSUE TITLE
MATLAB API for writing extension specifications

### ISSUE THREAD

**Issue by bendichter** [NeurodataWithoutBorders/matnwb#132]
Currently, if a MATLAB user wants to create an extension, they must write the YAML files by hand. pynwb supplies an API for automatically generating these files, which makes the process of creating extensions easier and ensures that they are well-formed. It would be nice to have an API in MATLAB as well for writing NWB extension YAML files in the hdmf schema language.

**Comment by oruebel** [comment]
I agree that this would be useful, but I also think that this is mostly a would-be-nice-to-have feature. My main concern is that the effort required (both in terms of initial implementation and maintenance cost) is high compared to what we would gain.

**Comment by bendichter** [comment]
Yeah, I agree. I made this "P-low" and put it at the bottom of the feature stack [here](https://github.com/NeurodataWithoutBorders/matnwb/projects/1)
